### PR TITLE
Add support for encoding required to use AlephBFT

### PR DESCRIPTION
**Pull request by prince-chrismc**
## Description

Towards #322 just another step converting the POC/Hackathon before we can add the other half of this puzzle together

This folds in the https://github.com/aleph-zero-foundation/AlephBFT#dependencies requirement of the `paritytech-scale-codec` and this is a foundational step

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [x] I've requested a review  from "pyrsia/collaborators"

### Code Contributions

- [x] I've built the code `cargo build --workspace`.
- [x] I've run the unit tests `cargo test --workspace`.

### CODE REVIEW

File: src/blockchain/Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-multihash = {version = "0.16.1", features = ["serde-codec", "scale-codec"]}
+multihash = {version = "=0.16.0", features = ["serde-codec", "scale-codec"]}
```
Comment (prince-chrismc) [inline src/blockchain/Cargo.toml:7]: This needs to be the exact version. because it was bumped passed a major version 

https://github.com/multiformats/rust-multihash/commit/81215bb6427f069fddf6fafb40a126548e1aefa9

This does not play well with other deps that have not moved to the newest version

File: src/blockchain/src/crypto/hash_algorithm.rs [hunk 2]
```diff
@@ -14,10 +14,11 @@
    limitations under the License.
 */
 
+use codec::{Decode, Encode};
 use multihash::{Code, Multihash, MultihashDigest};
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, Hash, Decode, Encode)]
```
Comment (fishseabowl) [inline src/blockchain/src/crypto/hash_algorithm.rs:21]: If possible, can we support only one encoding and decoding format?

**Review by fishseabowl — APPROVED** [review]
LGTM 😄

File: src/blockchain/src/crypto/hash_algorithm.rs (hunk 2 above)
Comment (prince-chrismc) [reply src/blockchain/src/crypto/hash_algorithm.rs:21]: We talked about this in the office, there are never mixed (in our dependencies) so there should not be any problems.
However human error is always possible, we can write bad code but that's always true 😆 

TL;DR
- SCALE codec - for consensus
`serde_json` - for everything else

This is get revised when we know more

**Review by mseabornIBM — APPROVED** [review]
With my limited knowledge of the blockchain code it seems reasonable.
